activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -242,7 +242,16 @@ def references!(*table_names) # :nodoc:
     #   Model.select(:field).first.other_field
     #   # => ActiveModel::MissingAttributeError: missing attribute: other_field
     def select(*fields)
-      return super if block_given?
+      if block_given?
+        if fields
+          ActiveSupport::Deprecation.warn(<<-WARNING.squish)
```

Comment:
I added the deprecation because I thought the behavior of sending args and block at the same time was working just accidently, but happy to get more feedback on this. cc @sgrif